### PR TITLE
fix: refined text input label truncation for single-word overflow

### PR DESCRIPTION
**Pull request by harsh9975**
image
image Previously, single words exceeding a specified width of 120px were truncated, potentially impacting readability and aesthetics. Added new prop for wordBreak in BaseText and also expose it in the Text component.

### CODE REVIEW

File: packages/blade/src/components/Typography/BaseText/types.ts [hunk 1]
```diff
@@ -56,6 +56,7 @@ export type BaseTextProps = {
   fontStyle?: 'italic' | 'normal';
   textDecorationLine?: 'line-through' | 'none' | 'underline';
   lineHeight?: keyof Theme['typography']['lineHeights'];
+  wordBreak?: 'normal' | 'break-all' | 'keep-all' | 'break-word' | undefined;
```
Comment (anuraghazra) [inline packages/blade/src/components/Typography/BaseText/types.ts:59]: You can remove the `undefined` we are already setting this type optional `?:`

**Review by anuraghazra — DISMISSED** [review]

File: packages/blade/src/components/Typography/BaseText/types.ts (hunk 1 above)
Comment (snitin315) [reply packages/blade/src/components/Typography/BaseText/types.ts:59]:
```suggestion
  wordBreak?: 'normal' | 'break-all' | 'keep-all' | 'break-word';
```

**Review by anuraghazra — DISMISSED** [review]

**Comment by harsh9975** [comment]
@anuraghazra if you require any changes from my end do tag me.

**Review by snitin315 — APPROVED** [review]

**Comment by harsh9975** [comment]
finally merged🥳
honest-work-meme-cb0f0fb2227fb84b77b3c9a851ac09b095ab74d8-s1100-c50

**Comment by harsh9975** [comment]
hello @saurabhdaware @anuraghazra @snitin315 
Just saw razorpay is hiring Product Development Engineer [link](https://razorpay.com/jobs/jobs-all/detail/?gh_jid=4347044005).

Would be glad if you would like to refer me based on my skills.
Strong Hans-on in Java, Javascript, Typescript, react,nodejs and possess strong problem solving skills using Data Structures and algorithms and above beginner level understanding of HLD,LLD.
I have actively contributed to Microsoft, Razorpays open source projects.

https://github.com/microsoft/fluentui/pulls?q=is%3Apr+author%3Aharsh9975
https://github.com/formbricks/formbricks/pulls?q=is%3Apr+author%3Aharsh9975+
https://github.com/razorpay/blade/pulls?q=is%3Apr+author%3Aharsh9975+

This is just a short hand of me, you can get sound data point from my [profile](https://harish-gautam-portfolio.vercel.app/) and my [resume](https://harish-gautam-portfolio.vercel.app/assets/Harish_Gautam_Resume_SDE.pdf)  . Awaiting you respond. Have a nice day

102624618_133953194972318_8546996589500091313_n

**Comment by snitin315** [comment]
@harsh9975 Unfortunately, the position is no longer open.

**Pull request merged**